src/tests/functional/plugin/gpu/concurrency/gpu_concurrency_tests.cpp

```diff
@@ -189,6 +189,68 @@ TEST(canSwapTensorsBetweenInferRequests, inputs) {
     }
 }
 
+TEST(smoke_InferRequestDeviceMemoryAllocation, usmHostIsNotChanged) {
+    auto fn = ngraph::builder::subgraph::makeDetectionOutput(ngraph::element::Type_t::f32);
+
+    auto ie = ov::Core();
+    auto compiled_model = ie.compile_model(fn, CommonTestUtils::DEVICE_GPU);
+
+    ov::InferRequest infer_request1 = compiled_model.create_infer_request();
+    ov::InferRequest infer_request2 = compiled_model.create_infer_request();
+
+    auto input_tensor1 = infer_request1.get_input_tensor();
+    FuncTestUtils::fill_tensor(input_tensor1, 20, 0, 1, 0);
+
+    auto output_tensor1 = FuncTestUtils::create_and_fill_tensor(compiled_model.output().get_element_type(), compiled_model.output().get_shape());
+    auto output_tensor2 = infer_request2.get_output_tensor();
+
+    // Use tensor from infer request #2 as an output for infer request #1
+    infer_request1.set_output_tensor(output_tensor2);
+    ASSERT_NO_THROW(infer_request1.infer());
+
+    // Modify tensor somehow and save as a reference values
+    FuncTestUtils::fill_tensor(output_tensor2);
+
+    std::vector<float> ref_values;
+    ref_values.resize(output_tensor2.get_byte_size());
+    std::memcpy(ref_values.data(), output_tensor2.data(), output_tensor2.get_byte_size());
+
+    // Perform second infer() call with a system host memory tensor
+    infer_request1.set_output_tensor(output_tensor1);
+    ASSERT_NO_THROW(infer_request1.infer());
+
+    // Expect that output_tensor2 will not change it's data after infer() call
+    auto thr = FuncTestUtils::GetComparisonThreshold(InferenceEngine::Precision::FP32);
+    FuncTestUtils::compareRawBuffers(ref_values.data(),
+                                     output_tensor2.data<float>(),
+                                     ref_values.size(),
+                                     ov::shape_size(output_tensor2.get_shape()),
+                                     thr);
+}
+
+TEST(smoke_InferRequestDeviceMemoryAllocation, canSetSystemHostTensor) {
+    auto fn = ngraph::builder::subgraph::makeDetectionOutput(ngraph::element::Type_t::f32);
+
+    auto ie = ov::Core();
+    auto compiled_model = ie.compile_model(fn, CommonTestUtils::DEVICE_GPU);
+
+    ov::InferRequest infer_request1 = compiled_model.create_infer_request();
+    ov::InferRequest infer_request2 = compiled_model.create_infer_request();
+
+    auto input_tensor1 = infer_request1.get_input_tensor();
+    FuncTestUtils::fill_tensor(input_tensor1, 20, 0, 1, 0);
+
+    auto output_tensor1 = FuncTestUtils::create_and_fill_tensor(compiled_model.output().get_element_type(), compiled_model.output().get_shape());
+    auto output_tensor2 = infer_request2.get_output_tensor();
+
+    infer_request1.set_output_tensor(output_tensor2);
+    ASSERT_NO_THROW(infer_request1.infer());
+
+    FuncTestUtils::fill_tensor(input_tensor1, 10, 0, 1, 1);
+    infer_request1.set_output_tensor(output_tensor1);
+    ASSERT_NO_THROW(infer_request1.infer());
```

Comment:
Hi sshly, thanks for the clarify and I understand. Regarding the coredump issue we talked offline yesterday, I suppose there are 2 aspects to cover:
1) the hw targets, as seems the issue first not reproducible in your env but on some other targets(I cannot tell the exact difference)
2) the async call obviously trigger some corner cases which leads to the corrupted stack on some hw targets, while the sync call seems always perform good from  my test, and that's why I think it's better to cover threading async call as well